src/app/sessions/page.tsx

```diff
@@ -19,18 +20,32 @@ import { Heading } from "@/components/Heading";
 import { MEMBER_MAX_SESSIONS, NON_MEMBER_MAX_SESSIONS } from "@/lib/constants";
 import { useCartStore } from "@/stores/useCartStore";
 
-const remainingSessions = 11;
-const isMember = true;
-const firstName = "David";
-
-const maxSessions = isMember ? MEMBER_MAX_SESSIONS : NON_MEMBER_MAX_SESSIONS;
+//const remainingSessions = 11;
+//const isMember = true;
+//const firstName = "David";
 
 export default function SelectSessionPage() {
   const { push } = useRouter();
 
   const sessionsSelected = useCartStore((state) => state.cart.length);
 
   const [shake, setShake] = useState(false);
+  const [firstName, setFirstName] = useState(null);
+  const [isMember, setIsMember] = useState(false);
+  const [remainingSessions, setRemainingSessions] = useState(false);
+  const maxSessions = isMember ? MEMBER_MAX_SESSIONS : NON_MEMBER_MAX_SESSIONS;
+
+  function getUserInfo(id: string) {
+    axios.get(`http://localhost:3000/api/user/${id}`).then((response) => {
+      const data = response.data;
+      setFirstName(data.firstName);
+      setIsMember(data.member);
+      setRemainingSessions(data.remainingSessions);
+      console.log(data);
+    });
+  }
```

Comment:
use the fetch api instead, its more optimised for nextjs.

one thing to note is that the fetch api caches get requests by default in nextjs, you will have to set {cache: "no-store"} to disable caching